packages/repository/src/relations/belongs-to/belongs-to-accessor.ts

```diff
@@ -53,6 +53,10 @@ export function createBelongsToAccessor<
     const primaryKey = meta.keyTo;
     const sourceModel = await sourceRepository.findById(sourceId);
     const foreignKeyValue = sourceModel[foreignKey as keyof Source];
+    // different dbs use different empty values
+    if (foreignKeyValue === undefined || foreignKeyValue === null) {
```

Comment:
maybe use a falsy check like `if(!foreignKeyValue) {}`?

Reply:
👍 

@jannyHou  Another thing is, we didn't notice that we already have #2333 opened. #4147 is duplicate. I don't know if I should continue on this PR. Because this 'fix' will change some behavior of some connectors ( please the proposal above). 
So I am thinking to close this PR without merging it. Thoughts?

Reply:
@agnes512 I think #2333 and #4147 are talking about different issues:

2333: means when CREATE related item, if you provide a non-existing fk(like you have users with id 1 through 10, but you create an order for user with id=11), the record will still be created, which is wrong. The solution would be checking a fk's corresponding source model instance exists in db first, then perform the creation. The story is more like an improvement of our design.

4147: looks like a bug, when GET related item, the included items are wrong. like a todo item with id=1 doesn't belong to any todoList, which means it has an empty fk, but GET todos/1/todoList returns a todo list...and please note the WRONGLY returned todo list does exist in the db, the problem is it doesn't own the todo specified in the request path...

Reply:
Back to this PR, since the `belongsTo` relation only has a `get` method(because child only has the get access to its parent, but cannot modify it), I am good with your approach here :) just return an empty result when fk is missing. 

Something you may want to double check: should it return an empty ARRAY or just null/undefined?

Reply:
You're right. I think the belongsTo accessor only checks the existence. And hasMany and hasOne do creation and traverse.

As for the return, I think this workaround should match with other connectors to throw an error: `with id "constraint {\"id\":null}"`

Once we done the memory connector referential integrity, this workaround should be removed, and let each connector to handle it.

Reply:
> As for the return, I think `undefined` is better than `[]`.

In the PR description you said the connectors return `[]`, so should we try to be consistent with that?

Reply:
@nabdelgadir that was my proposal before. But I think Janny's suggestion makes more sense. Will update the description and the title later